processing/thermal.py

```diff
@@ -143,8 +186,40 @@ def update_metadata(conf, recording, api):
     api.update_metadata(recording, metadata, complete)
 
 
-def upload_tracks(api, recording, algorithm_id, tracks):
-    for track in tracks:
-        track["id"] = api.add_track(recording, track, algorithm_id)
-        if "tag" in track:
+def upload_tracks(api, recording, main_model, model_results=None):
```

Comment:
What about:
1.  Uploading all the main-model and tracks.   
  - always mark as live
  - same for both if models specified in classifier or in processor

Then...

2.  Uploading all secondary model track labels after that (one model at a time)?
  - always mark as not live
  - get track id from main track and copy accross and then just write to db as normal?